Issue82 - Added scripts to program

Lyskawa-Aonyx opened the pull request:
Fixes #82 

**What was changed?**
Added shell script and AppleScript to root directory to enhance user friendliness of program startup.

*Here, describe what part of the application you changed (e.g. login page, database, etc.). If possible, mention what specific components were added, removed, or modified.*
Added esp.sh file to the program.
Added ESP.app to the program

**Why was it changed?**
It was changed based on customer feedback on their frustrations with getting the program to run before. 

*Here, describe the issue that you are fixing with this code, and why your code fixes it.*
I added the esp.sh program to make it easier for the user to run the program. I then made an AppleScript application that the user just needs to click on in order to run the program. The AppleScript relies on the esp.sh to function. 

*How to test the implemented changes?*
To test, you need to open terminal and navigate to the root folder of the program where esp.sh resides. You need to execute the command "chmod +x esp.sh" and then you will be able to run the script using "./esp.sh". If you want to test the AppleScript application, you need to double-click on ESP and follow the permissions it asks for. Docker does not need to be running before starting either script.

SiriChandanaGarimella commented:
@Lyskawa-Aonyx - Great work! This adds great value to the client feedback. Here are some points to consider:

**Permissions**: Can you please review the permissions required? Many of the permissions listed may not be necessary for our specific use case. Please make sure that we request only the essential permissions to ensure user privacy.

**Create new issues for the following:**

**Application Termination**: We should create a new issue to develop a script that safely terminates the application and cleans up by removing all associated Docker images. This will ensure proper resource management and system cleanup.
**Cross-Platform Compatibility**: While the current implementation for macOS is suitable as the client is using a Mac laptop, we need to ensure future compatibility with Windows OS. We should create a new issue to enhance the script for cross-platform compatibility.

kate-holdener commented on ESP.app/Contents/MacOS/applet:
This is showing up as a 'binary' file. Was this added accidentally?

Lyskawa-Aonyx replied:
This was added intentionally for the end user (on Mac, I will be implementing a Windows version) to be able to easily open and close the program while it is in this 'beta' state. A much more elegant solution will be implemented for deployment.

kate-holdener commented on ESP.app/Contents/Resources/Scripts/main.scpt:
This is showing up as a binary file, was it added accidentally?

kate-holdener commented on ESP.app/Contents/Resources/applet.icns:
This is showing up as a binary file, was it added accidentally?

Lyskawa-Aonyx commented:
I have pushed a new version of ESP.app. I have implemented a preliminary termination behavior. I have also reviewed all of the app permissions and they are currently as minimal as I can make them. I have also fixed the sleep timer to instead check if the "docker compose up" command is finished before opening the user's browser.

Lyskawa-Aonyx commented on ESP.app/Contents/Resources/applet.rsrc:
This was added intentionally for the end user (on Mac, I will be implementing a Windows version) to be able to easily open and close the program while it is in this 'beta' state. A much more elegant solution will be implemented for deployment.

SiriChandanaGarimella commented on esp.sh:
```diff
@@ -0,0 +1,23 @@
+docker compose up & 
+
+#open app
+sleep 7
```
@Lyskawa-Aonyx - The comment says, 'I have also fixed the sleep timer to check if the "docker-compose up" command is finished before opening the user's browser.' However, I still see a sleep 7 in the code. Could you please verify if your latest changes have been pushed? If not, could you include this update as part of the stop script issue?

SiriChandanaGarimella commented:
Closing this PR as this work is now included in the new PR #96
